orb.yml

```diff
@@ -8,62 +8,89 @@ examples:
+        codescene-ci-cd: empear/codescene-ci-cd@x.y.z
       workflows:
         version: 2
         my-workflow:
           jobs:
             - codescene-ci-cd/run-delta-analysis:
-                # The context specifies values for:
-                #   CODESCENE_HOST_URL, CODESCENE_USER, CODESCENE_PASSWORD and GITHUB_API_TOKEN
                 context: codescene-ci-cd
-                codescene-user: ${CODESCENE_USER}
-                # CODESCENE_DELTA_ANALYSIS_URL is project specific. 
-                # The value it should be set to is available in the CodeScene project settings 
-                codescene-delta-analysis-url: ${CODESCENE_HOST_URL}${CODESCENE_DELTA_ANALYSIS_URL}
+
+  delta_analysis_with_custom_values:
+    description: >
+      Run CodeScene Delta Analysis.
+      Custom values for parameters are specified directly on the job.
+    usage:
+      version: 2.1
+      orbs:
+        codescene-ci-cd: empear/codescene-ci-cd@x.y.z
+      workflows:
+        version: 2
+        my-workflow:
+          jobs:
+            - codescene-ci-cd/run-delta-analysis:
+                codescene-delta-analysis-url: ${MY_DELTA_ANALYSIS_URL_ENV_VAR}
+                codescene-user: ${MY_CODESCENE_USER_ENV_VAR}
+                codescene-password: ${MY_CODESCENE_PASSWORD_ENV_VAR}
+                github-api-token: ${MY_GITHUB_API_TOKEN_ENV_VAR}
 
 jobs:
   run-delta-analysis:
     description: >
       Run CodeScene Delta Analysis.
-      Requires environment variables for the CodeScene password and GitHub API Token.
+      Requires setting the codescene-delta-analysis-url, codescene-user, codescene-password and github-api-token parameters,
+      either explicitly or by setting the corresponding environment variables using the CircleCI UI
     parameters:
       codescene-delta-analysis-url:
-        description: # The CodeScene delta analysis API URL
+        description: >
+          The CodeScene delta analysis API URL. 
+          Set the parameter or use the CircleCI UI to add the value to the 'CODESCENE_DELTA_ANALYSIS_URL' env variable.
         type: string
+        default: ${CODESCENE_DELTA_ANALYSIS_URL-not-set}
       codescene-user:
-        description: # The name of the codescene user
+        description: >
+          The name of a CodeScene user.
+          Set the parameter or use the CircleCI UI to add the value to the 'CODESCENE_USER' env variable.
+        type: string
+        default: ${CODESCENE_USER-not-set}
+      codescene-password:
+        description: >
+          The password for the specified CodeScene user
+          Set the parameter or use the CircleCI UI to add the value to the 'CODESCENE_PASSWORD' env variable.
         type: string
-      codescene-password-key:
-        description: # The name of the environment variable containing the codescene password
-        type: env_var_name
-        default: CODESCENE_PASSWORD
+        default: ${CODESCENE_PASSWORD-not-set}
```

Comment:
The way I used env_var_name I saw no benefit as compared to just doing this.
I hadn't really understood that environment variables are resolved after config processing and thought I should use env_var_name to make sure "secret" things are not leaked into logs and such. The end result appears to me to be exactly the same, but just having a string is in this case actually more flexible since a value can then be specified directly in the config if one should wish to do so. (Correct me if I'm wrong here)

Reply:
correct @knorrest but the idea is to _force_ people to _never_ input a sensitive value directly in their config, by removing the option to do so

Reply:
if you use `type: string`, even with a sensible env var-like default, folks could still commit a password to their VCS, which is obviously an antipattern we want to engineer against as much as we reasonably can

Reply:
Ok, fair enough, I'll revert the changes for password and token to use env_var_name.